File: java/Exercise2.java
// Exercise 2: Counting the Number of Characters
import java.util.Scanner;


public class Exercise2 {

	public static void main(String[] args) {
		// Create the Scanner object.
		Scanner sc = new Scanner(System.in);
		
		// Print the prompt asking the user for the input string.
		System.out.print("What is the input string? ");
		
		// Read in the input from the Scanner.
		String input = sc.nextLine();
		
		/*
		 * Get the length of the String above (.length()) and store it to an
		 * integer value (int).
		 */
		int inputLength = input.length();
		
		/* 
		 * Print out the message. One easy way to combine Strings in Java is
		 * with the '+' operator (called concatenation). Strings can
		 * readily be concatenated with other String objects and numeric values. 
		 */
		System.out.println(input + " has " + inputLength + " characters.");
		
		// Close the Scanner object.
		sc.close();
	}

}

File: py/Exercise2.py
#Exercise 2: Counting the Number of Characters

#Ask the user for the input string:
input_string = input("What is the input string? ")

#Use the len() method to count the length of the string and print it along with the user's input in a sentence:
#Note the use of %s for input_string and %d for len(input_string). %d is used for numbers.
print("%s has %d characters." % (input_string, len(input_string)))
